trace/kafka/kafka.go

```diff
@@ -74,18 +79,18 @@ func NewAsyncProducer(brokers []string, oo ...OptionFunc) (*AsyncProducer, error
 	cfg := sarama.NewConfig()
 	cfg.Version = sarama.V0_11_0_0
 
-	ap := AsyncProducer{cfg: cfg, chErr: make(chan error), tag: opentracing.Tag{Key: "type", Value: "async"}}
+	ap := AsyncProducer{cfg: cfg, chErr: make(chan error), tag: opentracing.Tag{Key: "type", Value: "async"}, enc: json.Encode}
 
 	for _, o := range oo {
 		err := o(&ap)
 		if err != nil {
-			return nil, err
+			return nil, errors.Errorf("Could not apply OptionFunc '%v' to producer : %v", runtime.FuncForPC(reflect.ValueOf(o).Pointer()).Name(), err)
```

Comment:
This looks really cryptic and is hard to understand (at least for me). Is there any way to refactor it?
What exactly are you trying to do?

Reply:
I was trying to return the name of the OptionFunc that failed; like we're already doing [here](https://github.com/beatlabs/patron/blob/26338ecddfc81993253c79f5ee2554aa1421c2eb/async/kafka/kafka.go#L131).

I see that the Option will return an error that is useful for debugging, and this is not truly needed.